src/bootstrap/doc.rs

```diff
@@ -446,6 +446,13 @@ impl Step for Std {
 
         t!(fs::copy(builder.src.join("src/doc/rust.css"), out.join("rust.css")));
 
+        let content = fs::read_to_string(builder.src.join("src/doc/version-switcher.js")).unwrap();
+        fs::write(
+            out.join("version-switcher.js"),
+            content.replace("/* VERSION TO BE REPLACED */", &builder.version),
```

Comment:
To be noted that this `builder.version` is "broken" because it returns `1.65.0` whereas the current version hosted is `1.63.0`. Not sure how to get around that either...

I'm thinking about doing it like this: if it's nightly, I reduce the number of version by 2, if it's beta by 1 and it's stable, I don't change it. Does it seem like the right way to do it?

Reply:
I'm confused -- when we're shipping nightly, the current version is 1.65.0? Why would this need to get offset?

My expectation is that new versions of the script are/will be picked up in doc.rust-lang.org/nightly/ after each nightly release, in doc.rust-lang.org/beta/ after each beta release, and both (doc.rust-lang.org/stable/ and doc.rust-lang.org/1.x.y/) after each stable release.

We can always have a toggle for switching to beta + nightly and have a separate replace here which puts the current channel (so that the script knows that the current version and that channel are the same).

Reply:
The problem is that there is no `1.65.0/std` link currently. So the version number, even if accurate, is wrong.

Reply:
Right, we would need to use the channel and the version number to compute the right link. I think we can assume that if the channel is nightly the current and current-1 versions should link to /nightly/ and /beta/ respectively, and prior versions can use the 'named/versioned' (1.x.0) URLs.